### PR TITLE
Adding database support + fixing issues with the backend

### LINKED ISSUE #12: Create A database.
### General Troubleshooting

- [X] I have checked for similar issues on the Issue-tracker.
- [X] I have checked the branches or the maintainers' PRs for upcoming features.

### Feature Request

Regarding the database system I will be using mariadb and jooq. Jooq will basically generate the database classes into java so they can be used with the bot. The database will be used to recorded bans, warns, kicks, etc. and will be use for the auto moderation system.

### Example Use-Case

```java
Null
```

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
As told in the issue #12 I will be adding database support.

Also closes #12

### CODE REVIEW

**Comment by Haleloch** [comment]
What is jooq ?

**Comment by RealYusufIsmail** [comment]
> What is jooq ?

https://www.jooq.org

Basically it generates the database table into java

File: src/main/java/io/github/org/programming/database/ModerationDatabase.java [hunk 1]
```diff
@@ -8,9 +8,9 @@
 
 public class ModerationDatabase {
 
-    private static final Logger logger = Database.logger;
+    private static final Logger logger = Database1.logger;
 
-    private static final Connection connection = Database.connection;
+    private static final Connection connection = Database1.connection;
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/database/ModerationDatabase.java]: Database1 ?

File: src/main/java/io/github/org/programming/database/Database1.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public class Database1 {
+    static final Logger logger = LoggerFactory.getLogger(Database1.class);
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/database/Database1.java]: Default log file ?

File: build.gradle [hunk 3]
```diff
@@ -136,6 +144,9 @@ task downloadDependencies {
     }
 }
 
+//def props = new Properties()
+//file(".env").withInputStream { props.load(it) }
+
```
Comment (Haleloch) [inline build.gradle]: Commented code ?

File: build.gradle [hunk 4]
```diff
@@ -145,22 +156,19 @@ jooq {
             generationTool {
                 logging = org.jooq.meta.jaxb.Logging.WARN
                 jdbc {
-                    driver = 'org.mariadb.jdbc.Driver'
-                    //ُTODO: Make this configurable
-                    url = System.getenv("JDBC_URL")
-                    user = System.getenv("DATA_BASE_USER_NAME")
-                    password = System.getenv("DATA_BASE_PASSWORD")
+                    driver = 'org.postgresql.Driver'
+                    url = "jdbc:postgresql://localhost:5432/postgres"
```
Comment (Haleloch) [inline build.gradle]: Maybe we can store it in env var ?

File: src/main/java/io/github/org/programming/backend/handler/SlashCommandHandler.java [hunk 5]
```diff
@@ -110,22 +112,27 @@ private void onFinishedRegistration() {
     public void onSlashCommandInteraction(@NotNull SlashCommandInteractionEvent event) {
         final SlashCommandExtender cmd = slashCommand.get(event.getName());
 
-        if (cmd.isOwnerOnly() && event.getUser().getIdLong() != botOwnerId()) {
+        if (cmd.build().isOwnerOnly() && event.getUser().getIdLong() != botOwnerId()) {
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/backend/handler/SlashCommandHandler.java:115]: I don't understand what is `build` for

File: src/main/resources/sql/V1__moderation.sql [hunk 6]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS moderation
```
Comment (Haleloch) [inline src/main/resources/sql/V1__moderation.sql]: Maybe we can add timestamp, what did you think ?

File: src/main/java/io/github/org/programming/database/Database.java [hunk 7]
```diff
@@ -52,11 +45,10 @@ private static void executeTableUpdate() {
             for (File file : Objects.requireNonNull(listOfFiles, "List of files is null")) {
                 if (file.isFile()) {
                     logger.info("Executing table update: {}", file.getName());
-                    statement.execute(file.getName());
+                    statement.executeUpdate(Files.readString(Path.of(file.getPath())));
                 }
             }
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/database/Database.java]: listOfFiles ?
code handles files?

File: src/main/java/io/github/org/programming/database/Database.java [hunk 8]
```diff
@@ -7,40 +7,33 @@
...
-        config.addDataSourceProperty("prepStmtCacheSize", "250");
-        config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
-        config.addDataSourceProperty("characterEncoding", "utf8");
-        config.addDataSourceProperty("useUnicode", "true");
-        config.setMaximumPoolSize(10);
+        Properties props = new Properties();
+        props.setProperty("dataSourceClassName", "org.postgresql.ds.PGSimpleDataSource");
+        props.setProperty("dataSource.databaseName", "postgres");
+        props.setProperty("dataSource.portNumber", String.valueOf(5432));
+        props.setProperty("dataSource.serverName", "localhost");
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/database/Database.java]: get data from the env file

File: src/main/java/io/github/org/programming/backend/handler/SlashCommandHandler.java (hunk 5 above)
Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/backend/handler/SlashCommandHandler.java:115]: Have a look at the example command. It i used to build the command, like name, description, adding options ,etc

File: src/main/resources/sql/V1__moderation.sql (hunk 6 above)
Comment (RealYusufIsmail) [reply src/main/resources/sql/V1__moderation.sql]: yeah sure

File: build.gradle (hunk 4 above)
Comment (RealYusufIsmail) [reply build.gradle]: I had issue with that in gradle. It causes an error with the work flow

File: build.gradle (hunk 3 above)
Comment (RealYusufIsmail) [reply build.gradle]: ignore it.

File: src/main/java/io/github/org/programming/database/Database.java (hunk 8 above)
Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/database/Database.java]: fixed

File: src/main/java/io/github/org/programming/database/Database.java (hunk 7 above)
Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/database/Database.java]: what else should I do?

File: build.gradle (hunk 4 above)
Comment (RealYusufIsmail) [reply build.gradle]: Fixed

File: src/main/java/io/github/org/programming/database/Database.java (hunk 7 above)
Comment (Haleloch) [reply src/main/java/io/github/org/programming/database/Database.java]: which files are read ?

Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/database/Database.java]: oh, see src/main/resources/sql. The sql file and the future ones are read

**Review by Haleloch — APPROVED** [review]
PR ok

**Pull request merged**